### PR TITLE
test/e2e: check MC deployed on every node

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@linux.com>

**- What I did**

TestMCDeployed wasn't checking that the MC was indeed deployed on every
node (worker). Fix that by checking every node. The test is going to
take a little longer but it's more robust as well.

**- How to verify it**

run e2e

**- Description for the changelog**

### CODE REVIEW

File: test/e2e/mcd_test.go [hunk 1]
```diff
@@ -118,25 +120,39 @@ func TestMCDeployed(t *testing.T) {
...
-				if strings.Contains(line, "completed update for config "+newMCName) {
-					return true, nil
+			for _, pod := range mcdList.Items {
+				res, err := k.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &v1.PodLogOptions{}).DoRaw()
+				if err != nil {
+					// do not error out, we may be rebooting, that's why we list at every iteration
+					continue
+				}
+				for _, line := range strings.Split(string(res), "\n") {
+					if strings.Contains(line, "completed update for config "+newMCName) {
```
Comment (cgwalters) [inline test/e2e/mcd_test.go]: Seems odd to parse the log files for this; why not verify the `currentConfig` annotation?

Comment (cgwalters) [reply test/e2e/mcd_test.go]: Or for that matter, watch the worker machineconfigpool, which is doing just that.

Comment (runcom) [reply test/e2e/mcd_test.go]: changed to watch for status changes for the pool (even if now it's intrinsically racy cause we may somehow be missing some status, for instance, we may very well miss the Updating status, or check the previous Updated status thinking all is good if we don't check Updating and so on).

Comment (runcom) [reply test/e2e/mcd_test.go]: ok changed to check `currentConfig` instead.

**Comment by runcom** [comment]
Haproxy flake

/retest

**Pull request merged**